src/Data/Csv/Parser.hs

```diff
@@ -134,22 +159,31 @@ removeBlankLines = filter (not . blankLine)
 -- CSV file is allowed to not have a terminating line separator. You
 -- most likely want to use the 'endOfLine' parser in combination with
 -- this parser.
-record :: Word8  -- ^ Field delimiter
+record :: Word8     -- ^ Field delimiter
+       -> Escaping  -- ^ Escape
        -> AL.Parser Record
-record !delim = V.fromList <$!> field delim `sepByDelim1'` delim
+record !delim !escape = V.fromList <$!> field delim escape `sepByDelim1'` delim
 {-# INLINE record #-}
 
--- | Parse a field. The field may be in either the escaped or
--- non-escaped format. The return value is unescaped.
-field :: Word8 -> AL.Parser Field
-field !delim = do
+-- | Parse a field. The escape option determines whether fields can be
+-- escaped with double quotes.
+field :: Word8     -- ^ Field delimiter
+      -> Escaping  -- ^ Escape
+      -> AL.Parser Field
+field !delim Escape = fieldEscape delim
+field !delim NoEscape = fieldNoEscape delim
+{-# INLINE field #-}
+
+-- | Parse a field that may be in either the escaped or non-escaped
+-- format. The return value is unescaped.
+fieldEscape :: Word8 -> AL.Parser Field
+fieldEscape !delim = do
```

Comment:
Do we still need to allow non-escaped fields when escaping is not explicitly disabled? I suppose the answer is yes, because I suppose we have to handle files where only some fields are escaped.

(Please don’t post an example from user data here, as this repository is public.)

Reply:
Yes, exactly. Enabling the escaping/quoting option means that fields *can* be escaped, but don't have to be.